### PR TITLE
DDF-05014 Add extensibility to settings pane

### PR DESCRIPTION
**Pull request by zta6**
#### What does this PR do?
Allow settings components to render children
Allow settings components to pass in their own onClick handlers

#### Who is reviewing it? 
@andrewzimmer 
@Corey-Collins 
@willwill96 
@hayleynorton 

#### Ask 2 committers to review/merge the PR and tag them here.
@adimka
@bdeining

#### How should this be tested?
Verify that all items in the settings pane still function as desired
Verify that there are no errors in the developer console
Test downstream

#### What are the relevant tickets?
Fixes: #5014 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/user-settings/user-settings.tsx [hunk 1]
```diff
@@ -25,6 +25,8 @@ export type SettingsComponent = {
   component: JSX.Element
   text: string
   icon: string
+  onClick?: () => void
+  children?: any
```
Comment (andrewkfiedler) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/user-settings/user-settings.tsx]: Might be worth typing children to 
```
children?: React.ReactNode
```

**Comment by zta6** [comment]
@andrewkfiedler yes, that's exactly what its for. 
Typed children to `React.ReactNode` as you suggested.

**Comment by zta6** [comment]
build now

**Comment by willwill96** [comment]
Hero successful
Made sure settings pane still successful
Tested downstream

**Pull request merged**